drivers/kscan/kscan_stmpe811.c

```diff
@@ -0,0 +1,440 @@
+		xy_raw = (touch_data[0] << 16)|(touch_data[1] << 8)|(touch_data[2] << 0);
+		row = xy_raw >> 12;
+		col = xy_raw & 0xFFF;
+
+		/* Clear FIFO status register to clear the buffer */
+		ret = i2c_reg_write_byte_dt(&config->bus, STMPE811_REG_FIFO_STA, 0x1);
+		ret = i2c_reg_write_byte_dt(&config->bus, STMPE811_REG_FIFO_STA, 0x0);
+
+		col -= 350;
+
+		if (row <= 3000) {
+			row = 3900 - row;
+		} else {
+			row = 3800 - row;
+		}
+
+		/* Normalize */
+		col = col/11;
+		row = row/15;
+
+		if (col <= 0) {
+			col = 1;
+		} else if (col >= 320) {
+			col = 319;
+		}
+
+		if (row <= 0) {
+			row = 1;
+		} else if (row >= 240) {
+			row = 239;
+		}
+
+		LOG_DBG("row: %d, col: %d", row, col);
+
+		/* Clear interrupt flags */
+		ret = i2c_reg_write_byte_dt(&config->bus, STMPE811_REG_INT_STA, 0xFF);
+	}
+
+	data->callback(dev, row, col, pressed);
+
+	return ret;
+}
+
+static void stmpe811_work_handler(struct k_work *work)
+{
+	struct stmpe811_data *data = CONTAINER_OF(work, struct stmpe811_data, work);
+
+	stmpe811_process(data->dev);
+}
+
+#ifdef CONFIG_KSCAN_STMPE811_INTERRUPT
+static void stmpe811_isr_handler(const struct device *dev,
+			       struct gpio_callback *cb, uint32_t pins)
+{
+	struct stmpe811_data *data = CONTAINER_OF(cb, struct stmpe811_data, int_gpio_cb);
+
+	k_work_submit(&data->work);
+}
+#else
+static void stmpe811_timer_handler(struct k_timer *timer)
+{
+	struct stmpe811_data *data = CONTAINER_OF(timer, struct stmpe811_data, timer);
+
+	k_work_submit(&data->work);
+}
+#endif
+
+static int stmpe811_enable_callback(const struct device *dev)
+{
+	struct stmpe811_data *data = dev->data;
+
+#ifdef CONFIG_KSCAN_STMPE811_INTERRUPT
+	const struct stmpe811_config *config = dev->config;
+
+	gpio_add_callback(config->int_gpio.port, &data->int_gpio_cb);
+#else
+	k_timer_start(&data->timer, K_MSEC(CONFIG_KSCAN_STMPE811_PERIOD),
+		      K_MSEC(CONFIG_KSCAN_STMPE811_PERIOD));
+#endif
+
+	return 0;
+}
+
+static int stmpe811_disable_callback(const struct device *dev)
+{
+	struct stmpe811_data *data = dev->data;
+
+#ifdef CONFIG_KSCAN_STMPE811_INTERRUPT
+	const struct stmpe811_config *config = dev->config;
+
+	gpio_remove_callback(config->int_gpio.port, &data->int_gpio_cb);
+#else
+	k_timer_stop(&data->timer);
+#endif
+
+	return 0;
+}
+
+static int stmpe811_configure(const struct device *dev,
+			    kscan_callback_t callback)
+{
+	struct stmpe811_data *data = dev->data;
+
+	if (!callback) {
+		LOG_ERR("Invalid callback (NULL)");
+		return -EINVAL;
+	}
+
+	data->callback = callback;
+
+	return 0;
+}
+
+static int stmpe811_init(const struct device *dev)
+{
+	const struct stmpe811_config *config = dev->config;
+	struct stmpe811_data *data = dev->data;
+
+	int ret = 0;
+	uint16_t chip_id = 0;
+	uint8_t cmd = 0;
+	uint8_t reg = 0;
+
+	data->dev = dev;
+	k_work_init(&data->work, stmpe811_work_handler);
+
+	if (!device_is_ready(config->bus.bus)) {
+		LOG_ERR("I2C controller device not ready");
+		return -ENODEV;
+	}
+
+	cmd = STMPE811_REG_CHIP_ID;
```

Comment:
What's going on here? Above, `cmd` is initialized to 0. Here, it's overwritten to this value, making the initialization a dead store. But the following line hard-codes STMPE811_REG_CHIP_ID again instead of reading `cmd`.